ScanShop.Mobile/ScanShop.Mobile/ViewModels/QRScanViewModel.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using System.Windows.Input;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+using ZXing;
+
+namespace ScanShop.Mobile.ViewModels
+{
+    public class QRScanViewModel : BaseViewModel
+    {
+        public Command ScanCommand { get; set; }
+
+        private string _QRScanText = "Scan the QR code";
+
+        public QRScanViewModel()
+        {
+            ScanCommand = new Command(ScanResultAction);
+        }
+
+        public string QRScanText
+        {
+            get => _QRScanText;
+            set
+            {
+                if (value == _QRScanText)
+                    return;
+                _QRScanText = value;
+                OnPropertyChanged();
+            }
+        }
+
+        private void ScanResultAction(object obj)
```

Comment:
In my opinion, the method name "ProcessScanResult" or "HandleScanResult" would be better

Reply:
Fixed